### ISSUE TITLE
add shorturl option to image view

### ISSUE THREAD

**Issue by autonome** [filecoin-project/slate#488]
# Purpose

http gateway urls already give up decentralization, so we should also provide short urls for easy sharing.

# Deliverable

* image view on slate.host now has a cid, gateway url and a short url option

* api response for objects in a slate should include short url

# Dependencies

N/A

**Comment by cwaring** [comment]
I'd love to discuss the motivation of this further. Personally I think it's a great to keep the CID in the url, this makes the object hash very accessible, aids with awareness of a CID and also provides a great way to browse manually if I already have a CID of an object that I want to find on Slate.

I know we could add some of that functionality back in other ways but it's a strong stance to give it pride of place in URL. I see a future where I could access a Slate object directly via `slate://{CID}` and the UI will show me the object and all the connections with boards and users 💭

**Comment by jimmylee** [comment]
Are we saying that we can make `slate://` happen? If so, that would be insanely cool. I can't wait for

`slate://username:slatename:cid`

how insane would that be

**Comment by cwaring** [comment]
Yeah, we can register applinks via a desktop/electron wrapper or iOS/Android (with a bit of effort) to enable direct deep links. 

It's probably still better to only expose http links and redirect for pure accessibility but just [as an example](https://blog.branch.io/deep-linking-from-url-and-uri-schemes-to-universal-links-app-links-and-beyond/) for how something like this would work and why thinking carefully about our URL structure early on is important 😁

**Comment by autonome** [comment]
To be clear: I'm not saying Slate should change any existing URLs. I'm asking for a short URL option to be added in user share contexts, like the properties pane or any share icon.

There's no future world where winning is people looking at CIDs all the time. Visible CIDs may play some transitional role as a visual affordance in some contexts, but the endgame is most certainly not us staring at them.

We live in an HTTP world, and will be in a transitional stage for a lonnnnng time. Given the positioning of Slate as a product using the lingua franca as an adoption strategy, getting more people using Slate > getting people to use Slate to look at CIDs. So a short URL for places where shorter URLs will encourage more clicking is good, eg text messages and other chat platforms.

There's been some user research by Google into the cognitive load of URLs (I cannot find this link, it's driving me bonkers), and we just barely scratched the surface in the [Browser Design Guidelines](https://github.com/ipfs/browser-design-guidelines) work, where we found that there's a sweet spot of trust/confidence when there's 1) semantic information in URLs and 2) they're not too long. SO, exactly the opposite of a CID address 😂 I'm super curious if there's been research done in the short URL world around these types of questions...

**Comment by cwaring** [comment]
Agree human urls > computer urls, in this case we will get shorter computer urls - I'm not entirely convinced that's winning much but I'm open to experiment. Personally I'd rather seed fewer variations of urls to save overhead of management down the line and preserve the fragile connectivity of the web, another reason I'm a CID fan.

Being a big url nerd I'd love to see that study if you can dig it up sometime! (my semantic web days run deep).

@akuokojnr made similar remarks about keeping existing urls and adding a short url endpoint as a secondary option. @jimmylee how do you see this working?

One problem I see is that short urls will be unique to a single instance of slate and its data index, where as cid urls could potentially be portable across instances. I don't want to over optimise too early but I know we're trying to push this dweb boat off the shore so it's worth considering how this could play out.

**Comment by autonome** [comment]
> shorter computer urls - I'm not entirely convinced that's winning much

Short URLs are an industry. I don't want to point solely at the simultaneous rise of short URL usage and mobile chat as a dominant human communication form second only to spoken language and assume anything based on correlation... so am very much interested in seeing if there's social science research on short URLs.

> (my semantic web days run deep).

Let's talk this out over 🍻 sometime. There's much to atone for, Chris.

> One problem I see is that short urls will be unique to a single instance of slate and its data index,
> where as cid urls could potentially be portable across instances.

If I go to Slate, log in, open one of my slates and click on a specific image, this is the URL:

https://slate.host/dietrich/birds

😐

We're already not really walking that walk. My point is that we *shouldn't* - it's not part of the core strategy for Slate... well, based on my super limited "I've only been a user"-derived understanding of that strategy 😅

**Comment by autonome** [comment]
https://twitter.com/dietrich/status/1362239575956066305
